tests/foreman/ui/test_settings.py

```diff
@@ -478,6 +476,24 @@ def test_negative_update_hostname_with_empty_fact():
 
     :expectedresults: Error should be raised on setting empty value for
         hostname_facts setting
-
-    :CaseAutomation: notautomated
     """
+    property_name = "Discovered"
+    discovery_config_default_param = { "discovery_hostname" : "" }
+    discovery_config_default_param = {
+        content: entities.Setting().search(query={'search': f'name='
+                                                            f'{content}'})[
+            0] for content in discovery_config_default_param
+    }
+    discovery_config_new_param = { "discovery_hostname" : ""}
+    with session:
+        try:
+            for discovery_content, discovery_content_value in \
+                    discovery_config_new_param.items():
+                discovery_update_response = session.settings.update(
+                    discovery_content, discovery_content_value)
+            assert discovery_update_response != None, "Empty string accepted"
+        finally:
```

Comment:
Do we care? If we have to restore previous value because it might interfere with another test, then this should have `@run_in_one_thread` decorator.

Also, this should probably be in pytest teardown fixture.

Reply:
As there is an API test which uses the exact same value, thus restoring the values.